lib/build/run-tasks.js

```diff
@@ -0,0 +1,152 @@
+const colour = require('nodemon/lib/utils/colour')
+const del = require('del')
+const fs = require('fs')
+const fse = require('fs-extra')
+const path = require('path')
+const chokidar = require('chokidar')
+const sass = require('sass')
+
+const extensions = require('../extensions/extensions')
+
+const buildConfig = require('./config.json')
+const { paths } = buildConfig
+
+const appSassPath = path.join(paths.assets, 'sass')
+const docsSassPath = path.join(paths.docsAssets, 'sass')
+const v6SassPath = path.join(paths.v6Assets, 'sass')
+
+const appCssPath = path.join(paths.public, 'stylesheets')
+const v6CssPath = path.join(paths.public, 'v6', 'stylesheets')
+
+const v6PublicPath = path.join(paths.public, 'v6')
+
+function generateAssets () {
+  clean()
+  sassExtensions()
+  generateSass(appSassPath, appCssPath)
+  generateSass(docsSassPath, appCssPath)
+  generateSass(v6SassPath, v6CssPath)
+  copyAssets(paths.assets, paths.public)
+  copyAssets(paths.docsAssets, paths.public)
+  copyAssets(paths.v6Assets, v6PublicPath)
+}
+
+function runTasks () {
+  generateAssets()
+  watch()
+  runNodemon()
+}
+
+function clean () {
+  // doesn't clean extensions.scss, should it?
+  del.sync(['public/**', '.port.tmp'])
+}
+
+function sassExtensions () {
+  let fileContents = '$govuk-extensions-url-context: "/extension-assets"; '
+  fileContents += extensions.getFileSystemPaths('sass')
+    .map(filePath => `@import "${filePath.split(path.sep).join('/')}";`)
+    .join('\n')
+  fs.writeFileSync(path.join('lib', 'extensions', '_extensions.scss'), fileContents)
+}
+
+function generateSass (sassPath, cssPath) {
+  if (!fs.existsSync(sassPath)) return
+  console.log('compiling CSS...')
+  fs.mkdirSync(cssPath, { recursive: true })
+  fs.readdirSync(sassPath).forEach(file => {
+    if (!file.endsWith('.scss')) return
+
+    try {
+      const result = sass.renderSync({
+        file: `${sassPath}/${file}`,
+        logger: sass.Logger.silent,
+        loadPaths: [__dirname],
+        sourceMap: true,
+        style: 'expanded'
+      })
+
+      const cssFilename = file.replace('.scss', '.css')
+      fs.writeFileSync(`${cssPath}/${cssFilename}`, result.css)
+    } catch (err) {
+      console.error(err.message)
+      console.error(err.stack)
+    }
+  })
+}
+
+function copyAssets (sourcePath, targetPath) {
+  if (!fs.existsSync(sourcePath)) return
+  console.log('copying assets...')
+  const filterFunc = (src) => !src.startsWith(path.join(sourcePath, 'sass'))
+
+  // shouldnt have to mkdir, but copy errors with EEXIST otherwise
```

Comment:
is this comment in the right place? I think I wrote it for a mkdir I had, but I don't see one here

Reply:
Comment changed to:
// shouldn't have to ensure the directory exists, but copy errors with EEXIST otherwise